podo/PodoCount_code/podocount_main_serv.py

```diff
@@ -0,0 +1,286 @@
+
+    #if ftype == '.ndpi':
+        #level = 2
+    #else:
+    level = 0 #changed form 1 to 0. Check levels later
+
+    if (len(levels) > 1):
+        level = 1
+    else:
+        level = 0
+
+    WSI_levels = WSI_file.level_dimensions[level]
+
+    print(WSI_levels, "WSI_Levels")
+
+    downsample_factor=16
+    df2 = np.sqrt(downsample_factor)
+
+    #create output directory
+    #wsi_roi_dir = roi_dir + WSI_name
+    wsi_roi_dir = roi_dir
+    #wsi_pod_seg_dir = pod_seg_dir + WSI_name
+    wsi_pod_seg_dir = pod_seg_dir
+
+    if not os.path.isdir(wsi_roi_dir):
+        os.mkdir(wsi_roi_dir)
+        os.mkdir(wsi_pod_seg_dir)
+
+    #get whole-slide glom mask
+    #WSI_glom_xml = cwd + '/glom_xmls/' + WSI_name + '.xml'
+    WSI_glom_xml = args.glom_xml
+    WSI_glom_mask = xml_to_mask(WSI_glom_xml, (0,0), (WSI_cols,WSI_rows), downsample_factor=downsample_factor, verbose=0)
+    WSI_glom_mask = np.array(WSI_glom_mask)
+
+    #get whole-slide mask
+    #WSI_downsample = np.array(WSI_file.read_region((0,0),level,(WSI_levels[0],WSI_levels[1])),dtype = 'uint8')
+    WSI_downsample = np.asarray(WSI_file.read_region((0,0),level, levels[level]),dtype = 'uint8')
+
+    WSI_downsample = get_wsi_mask(WSI_downsample[:,:,0:3],WSI_glom_mask)
+
+    #xml files - initiation
+    xml_counter = counters + WSI_name +'_counter.xml'
+    xml_contour = contours + WSI_name + '_contour.xml'
+
+    xml_counter = open(xml_counter,'w')
+    xml_contour = open(xml_contour,'w')
+
+    xml_counter.write('<Annotations>\n\t<Annotation Id="1">\n\t\t<Attributes>\n\t\t\t<Attribute Id="0" Name="Pod" Value="" />\n\t\t</Attributes>\n\t\t<Regions>')
+    xml_contour.write('<Annotations>\n\t<Annotation Id="1">\n\t\t<Attributes>\n\t\t\t<Attribute Id="0" Name="Pod" Value="" />\n\t\t</Attributes>\n\t\t<Regions>')
+
+    gcount = 0
+    pcount = 0
+
+    #get ROI coordinates
+    print('-- Step 1: Glomerulus localization and quantification --\n')
+    bbs, total_gloms, glom_feat_labels, glom_feat_qty, glom_feat_array = get_glom_props(WSI_glom_mask,WSI_downsample,num_sections,dist_mpp,area_mpp2,df2)
+
+    #define pod feature spaces
+    glom_pod_feat_labels, glom_pod_feat_qty, glom_pod_feat_array, indv_pod_feat_labels, indv_pod_feat_qty, indv_pod_feat_array = get_pod_feat_spaces(total_gloms)
+
+    print('-- Step 2: Podocyte detection and quantification --\n')
+    for bb in range(len(bbs)):
+        bb_iter = bb
+        bb = bbs[bb]
+        x_start,y_start,x_stop,y_stop = int(df2*bb[0]),int(df2*bb[1]),int(df2*bb[2]),int(df2*bb[3])
+        x_length = x_stop-x_start
+        y_length = y_stop-y_start
+
+        roi = np.array(WSI_file.read_region((y_start,x_start),0,(y_length,x_length)),dtype = 'uint8')
+        roi = roi[:,:,0:3]
+        rows,cols,dims = roi.shape
+        roi_name = '/roi_' + str(bb_iter+1) + '.png'
+        roi_path = wsi_roi_dir + roi_name
+        #roi_path = output_dir + roi_name
+        plt.imsave(roi_path,roi)

```

Comment:
Why are we saving this? Is it used somewhere?

Reply:
It's not being used anywhere else. I think they were saving them to check the outputs locally. Updated the code.